lib/postgrex/protocol.ex

```diff
@@ -1049,30 +1049,31 @@ defmodule Postgrex.Protocol do
     end
   end
 
-  defp recv_simple(s, status, columns, rows, tags, buffer) do
+  defp recv_simple(s, status, results, columns, rows, tags, buffer) do
     case msg_recv(s, :infinity, buffer) do
       {:ok, msg_row_desc(fields: fields), buffer} ->
         columns = column_names(fields)
-        recv_simple(s, status, columns, rows, tags, buffer)
+        recv_simple(s, status, results, columns, rows, tags, buffer)
 
       {:ok, msg_data_row(values: values), buffer} ->
         row = Types.decode_simple(values, s.types)
-        recv_simple(s, status, columns, [row | rows], tags, buffer)
+        recv_simple(s, status, results, columns, [row | rows], tags, buffer)
 
       {:ok, msg_command_complete(tag: tag), buffer} ->
-        recv_simple(s, status, columns, rows, [tag | tags], buffer)
+        result = done(s, status, columns, Enum.reverse(rows), Enum.reverse([tag | tags]))
+        recv_simple(s, status, [result | results], [], [], [], buffer)
```

Comment:
It seems to me that this function now never receives something other than `[]` as `tags`, so maybe the `tags` param could be removed altogether?
 
```suggestion
        result = done(s, status, columns, Enum.reverse(rows), [tag])
        recv_simple(s, status, [result | results], [], [], buffer)
```

Reply:
ty nice catch, updated